### PR TITLE
[G_HIU_69] Gayatri/Dheeraj | Implement /v1/patients/{id} API

### PR DESCRIPTION
**Pull request by gayatrijagtap**

### CODE REVIEW

File: src/main/java/in/org/projecteka/hiu/patient/model/Requester.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package in.org.projecteka.hiu.patient.model;
+
+import lombok.Value;
+
+@Value
+public class Requester {
```
Comment (angshu) [inline src/main/java/in/org/projecteka/hiu/patient/model/Requester.java]: implement serializable.

File: src/main/java/in/org/projecteka/hiu/patient/PatientService.java [hunk 2]
```diff
@@ -33,4 +49,38 @@
...
                                 }));
     }
+
+    public Mono<Patient> findPatientWith(String id) {
+        return getFromCache(id, () ->
+        {
+            var cmSuffix = getCmSuffix(id);
+            var request = getFindPatientRequest(id);
+            return centralRegistry.token()
+                    .flatMap(token -> scheduleThis(gatewayServiceClient.findPatientWith(request, token, cmSuffix))
```
Comment (angshu) [inline src/main/java/in/org/projecteka/hiu/patient/PatientService.java]: I feel we can easily push the token fetching to the GatewayServiceClient now. Does not need to be a concern of the PatientService or any other Service. HIU is only talking to Gateway now.

**Review by angshu — APPROVED** [review]

File: src/main/java/in/org/projecteka/hiu/patient/model/Requester.java (hunk 1 above)
Comment (gayatrijagtap) [reply src/main/java/in/org/projecteka/hiu/patient/model/Requester.java]: done

File: src/main/java/in/org/projecteka/hiu/patient/PatientService.java (hunk 2 above)
Comment (gayatrijagtap) [reply src/main/java/in/org/projecteka/hiu/patient/PatientService.java]: done

**Pull request merged**